### PR TITLE
feat: add kms key to auth secret

### PR DESCRIPTION
**Pull request by jjfrench**
## Description
- created policy document for KMS key (similar to key aws/secretsmanager)
- adds custom KMS key to secret
  - enables secret to be read across accounts

### CODE REVIEW

File: infra/AuthStack.py [hunk 1]
```diff
@@ -171,6 +171,89 @@ def _get_client_secret(
...
+                    actions=[
+                        "kms:Encrypt",
+                        "kms:Decrypt",
+                        "kms:ReEncrypt*",
+                        "kms:CreateGrant",
+                        "kms:DescribeKey"
+                    ],
+                    effect=iam.Effect.ALLOW,
+                    resources=["*"],
+                    principals=[iam.ArnPrincipal("*")],
```
Comment (hrodmn) [inline infra/AuthStack.py:192]: What does it mean to allow any principal for this policy?

File: infra/AuthStack.py [hunk 2]
```diff
@@ -171,6 +171,89 @@ def _get_client_secret(
...
+                    ],
+                    effect=iam.Effect.ALLOW,
+                    resources=["*"],
+                    principals=[iam.ArnPrincipal(f"arn:aws:iam::{Stack.of(self).account}:root")],
+                ),
+                iam.PolicyStatement(
+                    actions=["kms:Decrypt", "kms:DescribeKey"],
+                    effect=iam.Effect.ALLOW,
+                    resources=["*"],
+                    principals=[iam.ArnPrincipal("arn:aws:iam::884094767067:role/MAAP-ADE-K8S")],
```
Comment (hrodmn) [inline infra/AuthStack.py]: It would be helpful to set this as a constant with a descriptive name at the top.

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-aws-cdk-lib==2.35.0
+aws-cdk-lib==2.168.0
```
Comment (hrodmn) [inline requirements.txt:1]: :partying_face:

**Review by hrodmn — COMMENTED** [review]
Thanks for making this change! Just a few small comments but I think this looks right. It would be helpful to add a slightly more verbose description in the module to describe why we are adding this cross-account permission.

File: infra/AuthStack.py (hunk 1 above)
Comment (jjfrench) [reply infra/AuthStack.py:192]: AWS states it as "Allow access through AWS Secrets Manager for all principals in the account that are authorized to use AWS Secrets Manager"

File: infra/AuthStack.py (hunk 2 above)
Comment (jjfrench) [reply infra/AuthStack.py]: Ran through it a little fast, I'll add it to the env!

**Review by hrodmn — APPROVED** [review]
Thanks for adding the config var, this looks good!

**Pull request merged**